content/ctl_chef_solo.md

```diff
@@ -190,6 +190,19 @@ chef ALL=(ALL) NOPASSWD: /usr/bin/chef-solo
 where `chef` is the name of the non-root user. This would allow
 chef-solo to run any command on the node without requiring a password.
 
+{{< warning >}}
+
+The configuration for the 'chef' user as described above will still 
+going to have root-level access by calling system commands from a Chef 
+script file with sudo privileges. 
+
+To prevent such kind of security issues, it is highly recommended to create 
+an alternate approach for the cases where 'chef' user should not have 
+built-in root level access.
```

Comment:
```suggestion
Following the instructions above will effectively grant the `chef` user root-level access to system commands by allowing them to run arbitrary Ruby code in the form of Chef scripts with sudo privileges. To prevent security issues, it is highly recommended that this is only done on systems with trusted users.
```

I rewrote this with a few thoughts in mind:

* I tried to make the text more explicit about what the possible consequences are.
* Mentioning "an alternate approach" without saying what that might be feels incomplete.

Does this enable a certain use case, or make it more convenient? It might be worth adding information about why the user would want to do this so that users are not led to believe they need to do this.

Reply:
I think people implement such configuration might think to provide an option to run Chef scripts locally. To have a better approach can be proposed is to allow only to run "sudo chef-solo" with read-only Chef scripts where located outside of 'chef' user's home folder like
chef ALL=(ALL) NOPASSWD: /usr/bin/chef-solo -c /lib/scripts/chef.rb (where chef user only have read permission on this file)
Although, /lib/scripts/chef.rb should not be able to run any other set of commands where "chef" user can execute.